docker-images/kafka-based/kafka/scripts/kafka_readiness.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -e
+
+# Test ZK-based broker and KRaft broker readiness
+# The kafka-agent will create /var/opt/kafka/kafka-ready in the container when the broker
+# state is >= 3 && != 127 (UNKNOWN state)
+test -f /var/opt/kafka/kafka-ready
```

Comment:
Can the state `>= 3 && != 127` be reached without listening on the listeners? Also, in KRaft mode, if it would not be listening on port 9090, I assume the liveness probe will kill it. So that might be a second reason why is it not needed here.

The original assumption of the probes for sure was that the internal Kafka state and the liveness probe determine that listeners are alive and we do not need to check it again.

Reply:
The listening check below is to verify that the KRaft controller is ready. My understanding was that if this node was running as a controller only and not a broker, then the broker state metric would not be present. So that is the reason why we have two separate checks. At the moment we run both since we are running in combined mode, but in future we would have an if statement here that selects which checks to run based on the mode the node is running in.

Reply:
Sorry, but I think approach is really weird. I would expect this PR to do one of the following:
* Handle what we have today => in which case it should handle the combined nodes only
* Build for the future => in whcih case there should be a clear path for controller nodes, broker nodes and combined nodes. The paths for the controller and broker nodes will not be used today for anything, but we will need it one day so why not add it now. 

But instead this seems to ship some weird hybrid. We have code which doesn't distinguish between the node types. But it seems to have a code which is apparently there because one day we will have the node types. I think you need to pick one of those options and follow it.

So if this should follow the first option and create a solution only for the combined nodes today ... do we really expect the broker to reach one of the desired states but not listen on 9090?

If this delivers for the future, then there should be some clear code path which distinguishes the different modes and does only what is expected to be done on them.